projects/storefrontlib/feature-toggles.ts

```diff
@@ -26,5 +26,16 @@ export interface FeatureToggles {
      *    lazy loaded modules, it will have to use ConfigurationService.unifiedConfig$
      */
     disableConfigUpdates?: boolean;
+
+    /**
+     * To fix certain features when a customer agent is emulating a user, we must send a header called
+     * "sap-commerce-cloud-user-id" to the backend fetching certain data as the user, rather than the
+     * agent. However, the header will break instances of Commerce Cloud who do not allow requests with
+     * this header. (For example, the configuration "corsfilter.commercewebservices.allowedHeaders" will
+     * need to allow it.)
+     *
+     * Enabling this feature will send the emulated user's IDs on requests that need it as context.
+     */
+    enableCommerceCloudUserIdHeader?: boolean;
```

Comment:
I get the main reason for the feature flag is to prevent issues to customers when they upgrade to 5.1.  They would get errors form the requests that send the new header.
However it doesn't seem that this is a feature in of itself that customers want to toggle.

Do you think this behaviour could be hidden behind a 5.1 feature level flag instead?

In 6.0, it should/will become enabled by default.

Reply:
Aren't there cases where the customer upgrades Spartacus but does not upgrade Commerce Cloud?
This type of toggle seems more pragmatic.

Reply:
I'm not sure I understand.  Let's discuss.